src/topology/generation.rs

```diff
@@ -0,0 +1,151 @@
+use crate::internal_events::InternalReloadFailed;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+/// Generation counter incremented for each reload.
+static GENERATION: AtomicUsize = AtomicUsize::new(0);
+
+/// Increments generation causing all Ages to report that they
+/// are old.
+pub(super) fn inc_generation() {
+    GENERATION.fetch_add(1, Ordering::Relaxed);
+}
+
+/// Age to be paired with some relodable data.
+///
+/// Usually used through more generic wrapper
+/// structures, but can be used in custom structs
+/// where this general procedure should be followed:
+/// 1. Age::new()
+/// 2. create data
+/// 3. if Age::is_old() continue from 6.
+/// 4. reload data
+/// 5. if Age::set_age() is old repeat from 4.
+/// 6. use data
+///
+/// Start from 1. when first creating data.
+/// Start from 3. when accesing data.
+///
+/// Or update method can be used.
+#[derive(Debug, PartialEq, Eq)]
+pub struct Age {
+    gen: usize,
+}
+
+impl Age {
+    /// Creates new Age with current generation.
+    pub fn current() -> Self {
+        Age {
+            gen: GENERATION.load(Ordering::Relaxed),
+        }
+    }
+
+    /// Returns new age to be seted after the data
+    /// has been reloaded.
+    pub fn is_old(&self) -> Option<Age> {
+        let current = Self::current();
+        if current != *self {
+            Some(current)
+        } else {
+            None
+        }
```

Comment:
FWIW `bool` has a new method `then` that can shorten exactly this:
```suggestion
        (current != *self).then(|| current)
```